### PR TITLE
Fix sysrc version of service enabled check on FreeBSD.

### PR DESCRIPTION
**Pull request by koichirok**
##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

service module (system/service.py)
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel df81f1197f) last updated 2016/08/10 09:52:52 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 5d7b46e0dd) last updated 2016/08/10 09:53:01 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD eaa71f51d6) last updated 2016/08/10 09:53:01 (GMT +000)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

sysrc version of service enabled check fails in specific (but not special) situations. This PR fixes this issue.

For example, ansible fails sendmail's enabled check as follows when sendmail_msq_queue is set to NO,  enabled=yes and change is made:

```
$ grep sendmail /etc/rc.conf
sendmail_enable="NO"
sendmail_msp_queue_enable="NO"
$ ansible localhost -s -c local -m service -a 'name=sendmail enabled=yes'
127.0.0.1 | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "unable to set rcvar: sysrc did not change value", 
    "stderr": "", 
    "stdout": "sendmail_enable: NO -> YES\n", 
    "stdout_lines": [
        "sendmail_enable: NO -> YES"
    ]
}
```

With my PR, ansible not fail.

```
$ grep sendmail /etc/rc.conf
sendmail_enable="NO"
sendmail_msp_queue_enable="NO"
$ ansible localhost -s -c local -m service -a 'name=sendmail enabled=yes'
127.0.0.1 | SUCCESS => {
    "changed": true, 
    "enabled": true, 
    "name": "sendmail"
}
$ grep sendmail /etc/rc.conf
sendmail_enable="YES"
sendmail_msp_queue_enable="NO"
```

### CODE REVIEW

**Comment by koichirok** [comment]
I've refined a sentence in comment at 3ad947f.  only comment is changed.

File: system/service.py [hunk 1]
```diff
@@ -1009,8 +1009,10 @@ def service_enable(self):
                 if rc != 0:
                     self.module.fail_json(msg="unable to set rcvar using sysrc", stdout=change_stdout, stderr=change_stderr)
 
-                # sysrc does not exit with code 1 on permission error => validate successful change using service(8)
-                rc, check_stdout, check_stderr = self.execute_command("%s %s %s" % (self.svc_cmd, self.name, "enabled"))
+                # sysrc does not exit with code 1 on permission error => validate successful change using rc.subr(8) functions
+                # Due to the behavior of some scripts, we couldn't use service(8) here.
+                # For example, "service sendmail enabled" always fail if sendmail_msp_queue_enable is set to NO.
+                rc, check_stdout, check_stderr = self.execute_command("sh -c '. /etc/rc.subr;name=%s;rcvar=%s;load_rc_config $name;run_rc_command %s'" % (self.name, self.rcconf_key, "enabled"))
```
Comment (bcoca) [inline system/service.py]: you cannot always assume `sh` is available

Comment (koichirok) [inline system/service.py]: I've changed sh to /bin/sh. 

I think we can assume /bin/sh is always available at least current and earlier version of FreeBSD.

**Comment by koichirok** [comment]
I've changed sh call to /bin/sh. 
If it's acceptable, I'll 'git rebase -i' to merge two commits into one.

**Comment by gregdek** [comment]
Thanks @koichirok. To the current maintainers, @grossws please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @koichirok for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**